packages/entity-server/src/routes/hierarchy/extendedFieldFunctions.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { EntityType } from '../../models';
+
+const getParentCode = async (
+  entity: EntityType,
+  context: {
+    hierarchyId: string;
+  },
+) => {
+  const { hierarchyId } = context;
+  return (await entity.getParent(hierarchyId))?.code;
+};
+
+/**
+ * Faster than getParentCode() but requires pre-calculating the childToParentMap
+ * Use when performing getParentCode() for many entities
+ */
+export const getParentCodeBulk = (entity: EntityType, childToParentMap: Record<string, string>) => {
+  return childToParentMap[entity.code];
+};
```

Comment:
Honestly this seems like a function that adds more confusion than it solves? The content is super simple, and could be done and read inline. I would expect `getParentCodeBulk` to take a bulk list of entities and return all their parent codes.

Reply:
Yeah, you're right. Originally it contained a lot more logic here and made sense to be here but I might just switch to doing it all inline for these `bulk` operations 👍